app/src/main/java/com/babylon/wallet/android/presentation/main/MainScreen.kt

```diff
@@ -22,20 +22,28 @@ fun MainScreen(
     onNavigateToMnemonicRestore: () -> Unit,
     onAccountCreationClick: () -> Unit,
     onNavigateToOnBoarding: () -> Unit,
-    onNavigateToIncompatibleProfile: () -> Unit
+    onNavigateToIncompatibleProfile: () -> Unit,
+    onNavigateToRootDetectedContent: () -> Unit
 ) {
     val state by mainUiState.collectAsStateWithLifecycle()
     when (state.initialAppState) {
         is AppState.Wallet -> {
-            WalletScreen(
-                modifier = modifier,
-                viewModel = hiltViewModel(),
-                onMenuClick = onMenuClick,
-                onAccountClick = onAccountClick,
-                onAccountCreationClick = onAccountCreationClick,
-                onNavigateToMnemonicBackup = onNavigateToMnemonicBackup,
-                onNavigateToMnemonicRestore = onNavigateToMnemonicRestore
-            )
+            val initialWalletState = state.initialAppState as AppState.Wallet
+            if (initialWalletState.showDeviceRootedWarning) {
```

Comment:
was it not possible to make this dialog display snippet part of `WalletApp` and use it only once?

Reply:
> was it not possible to make this dialog display snippet part of `WalletApp` and use it only once?

I didn't think about it actually, I guess it will be possible. Do you believe it make more sense to do it from `WalletApp` ?

Reply:
benefit I see is that you would have this condition once, not twice in `when` clause. But it is not hard requirement

Reply:
@jakub-rdx Makes sense, pushed the fix